examples-gallery/plot_parameter_estimation.py

```diff
@@ -0,0 +1,205 @@
+eom = kd.col_join(fr + frstar)
+sm.pprint(eom)
+
+rhs = KM.rhs()
+
+qL = q_ind + u_ind
+pL = [m, c, k, l0]
+
+rhs_lam = sm.lambdify(qL + pL, rhs)
+
+
+def gradient(t, x, args):
+    return rhs_lam(*x, *args).reshape(8)
+
+t0, tf = 0, 20
+num_nodes = 500
+times = np.linspace(t0, tf, num_nodes)
+t_span = (t0, tf)
+
+x0 = np.array([3, 3, 3, 3, 0, 0, 0, 0])
+pL_vals = [1.0, 0.25, 1.0, 1.0]
+
+resultat1 = solve_ivp(gradient, t_span, x0, t_eval = times, args=(pL_vals,))
+resultat = resultat1.y.T
+
+noisy = []
+np.random.seed(123)
+for i in range(4):
+    noisy.append(resultat[:, i] + np.random.randn(resultat.shape[0]) * 0.5 +
+        np.random.randn(1)*2)
+
+fig, ax = plt.subplots(figsize=(10, 5))
+for i in range(4):
+    ax.plot(times, noisy[i], label=f'measurement {i+1}', lw=0.5)
+plt.xlabel('Time')
+ax.set_title('Measurements of the position of the mass')
+ax.legend();
+
+plt.show()
+
+
+# %%
+# Set up the Estimation Problem.
+# --------------------------------
+#
+# If some measurement is considered more reliable, its weight w can be increased.
+#
+# objective = :math:`\int_{t_0}^{t_f} (w_1 (x_1 - noisy_{x_1})^2 + w_2 (x_2 - noisy_{x_2})^2 + w_3 (x_3 - noisy_{x_3})^2 + w_4 (x_4 - noisy_{x_4})^2)\, dt`
+#
+state_symbols = [x1, x2, x3, x4, u1, u2, u3, u4]
+unknown_parameters = [c, k]
+
+interval_value = (tf - t0) / (num_nodes - 1)
+par_map = {m: pL_vals[0], l0: pL_vals[3]}
+
+w =[1, 1, 1, 1]
+def obj(free):
+    return interval_value *np.sum((w[0] * free[:num_nodes] - noisy[0])**2 +
+            w[1] * (free[num_nodes:2*num_nodes] - noisy[1])**2 +
+            w[2] * (free[2*num_nodes:3*num_nodes] - noisy[2])**2 +
+            w[3] * (free[3*num_nodes:4*num_nodes] - noisy[3])**2
+)
+
+
+def obj_grad(free):
+    grad = np.zeros_like(free)
+    grad[:num_nodes] = 2 * w[0] * interval_value * (free[:num_nodes] - noisy[0])
+    grad[num_nodes:2*num_nodes] = 2 * w[1] * (interval_value *
+                    (free[num_nodes:2*num_nodes] - noisy[1]))
+    grad[2*num_nodes:3*num_nodes] = 2 * w[2] * (interval_value *
+                    (free[2*num_nodes:3*num_nodes] - noisy[2]))
+    grad[3*num_nodes:4*num_nodes] = 2 * w[3] * (interval_value *
+                    (free[3*num_nodes:4*num_nodes] - noisy[3]))
+    return grad
+
+instance_constraints = (
+    x1.subs({t: t0}) - x0[0],
+    x2.subs({t: t0}) - x0[1],
+    x3.subs({t: t0}) - x0[2],
+    x4.subs({t: t0}) - x0[3],
+    u1.subs({t: t0}) - x0[4],
+    u2.subs({t: t0}) - x0[5],
+    u3.subs({t: t0}) - x0[6],
+    u4.subs({t: t0}) - x0[7],
+)
+
+problem = Problem(
+    obj,
+    obj_grad,
+    eom,
+    state_symbols,
+    num_nodes,
+    interval_value,
+    known_parameter_map=par_map,
+    instance_constraints=instance_constraints,
+    time_symbol=me.dynamicsymbols._t,
+)
+
+# %%
+# Initial guess.
+#
+initial_guess = np.array(list(np.random.randn(8*num_nodes + 2)))
+# %%
+# Solve the Optimization Problem.
+#
+solution, info = problem.solve(initial_guess)
+print(info['status_msg'])
```

Comment:
new cell before the plot command, basically a new cell should be after anything you print or plot